tooling/SecretManager.VisualStudio/SecretManagerVSPackage.cs

```diff
@@ -0,0 +1,17 @@
+// Copyright (c) .NET Foundation. All rights reserved.
+// Licensed under the Apache License, Version 2.0. See License.txt in the project root for license information.
+
+using System;
+using System.Runtime.InteropServices;
+using Microsoft.VisualStudio.Shell;
+
+namespace Microsoft.Extensions.SecretManager.VisualStudio
+{
+    [PackageRegistration(UseManagedResourcesOnly = true)]
+    [InstalledProductRegistration("#110", "#112", "1.0", IconResourceID = 400)]
+    [Guid(SecretManagerVSPackage.PackageGuidString)]
+    public sealed class SecretManagerVSPackage : Package
+    {
+        public const string PackageGuidString = "d6b82ee5-8f4b-41f2-a581-fefa7f57a239";
```

Comment:
Why do we need a VS package for this? That's not required for a MEF assembly. Is there some reason that's not obvious to me that we need this?

Reply:
Just following what File->New gave me and all other VSIX projects appear to have. This is my first VSIX and don't really understand how all this bootstrapping stuff works. If it's not required, I'll strip it out.

Reply:
Seems it's required. Without it, build fails:

Error		CreatePkgDef : error : ArgumentException: No Visual Studio registration attribute found in this assembly.
The assembly should contain an instance of the attribute 'Microsoft.VisualStudio.Shell.RegistrationAttribute' defined in assembly 'Microsoft.VisualStudio.Shell.Framework' version '15.0.0.0' 
   at Microsoft.VisualStudio.Tools.CreatePkgDef.ProcessAssembly(String fileName, Hive hive, PkgDefContext context, Boolean register, RegistrationMode mode)
   at Microsoft.VisualStudio.Tools.CreatePkgDef.DoCreatePkgDef(InputArguments inputArguments)
   at Microsoft.VisualStudio.Tools.CreatePkgDef.Main(String[] arguments)	SecretManager.VisualStudio

Reply:
I can help you try to figure this out tomorrow

Reply:
I'll take a look at this tomorrow with you and see if I can figure it out. @AArnott do you have an idea if there's something obvious we need to do besides delete the VSPackage and remove the entry from the vsixmanifest?

Reply:
There is a project property to disable pkgdef generation. 
However, you *do* want a pkgdef, since there is one VS registration attribute your assembly should always have:

```csharp
[assembly: ProvideCodeBase(CodeBase = @"$PackageFolder$\YOUR.ASSEMBLY.NAME.dll")]
```

That way, VS can find your assembly for sure instead of hoping. It improves overall reliability and even performance in loading your assembly. That should eliminate the pkgdef build error since you'll now have an attribute to create a pkgdef for. :)

Reply:
Awesome, thanks for the pointer. That worked.